smdebug/tensorflow/keras.py

```diff
@@ -947,14 +951,35 @@ def _save_layer_values(self, logs):
                 layer_name = layer_name.name
             elif isinstance(layer_name, bytes):
                 layer_name = str(layer_name, "utf-8")
+            layer_input_tensor_name = get_export_name_for_keras(str(layer_name), "input")
             if len(layer_input) == 1:
                 # Layer Inputs are flattened and passed as a list into
                 # the next layer. Unpacking it speeds up the _make_numpy fn.
                 layer_input = layer_input[0]
-            layer_input_tensor_name = get_export_name_for_keras(str(layer_name), "input")
-            self._save_tensor_to_file(layer_input_tensor_name, layer_input, collections_to_write)
+                self._save_tensor_to_file(
+                    layer_input_tensor_name, layer_input, collections_to_write
+                )
+            else:
+                idx = 0
+                for l_name in layer_input:
+                    layer_input_tensor_name_with_idx = f"{layer_input_tensor_name}_{idx}"
+                    self._save_tensor_to_file(
+                        layer_input_tensor_name_with_idx, l_name, collections_to_write
+                    )
+                    idx += 1
             layer_output_tensor_name = get_export_name_for_keras(str(layer_name), "output")
-            self._save_tensor_to_file(layer_output_tensor_name, layer_output, collections_to_write)
+            if isinstance(layer_output, list):
```

Comment:
nit: might be cleaner to have a single if check to convert layer_output to a list if it's not already a list.

```
if not isinstance(layer_output, list):
    layer_output = [layer_output]
for l_output in layer_output:
...
```

Reply:
Saving layer output values as a list using the _make_numpy fn is slow. Saving them as tensors when they're not lists improves performance